subprojects/scala/src/integTest/groovy/org/gradle/scala/compile/ScalaCompileRelocationIntegrationTest.groovy

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.scala.compile
+
+import org.gradle.integtests.fixtures.AbstractTaskRelocationIntegrationTest
+import org.gradle.scala.ScalaCompilationFixture
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+
+import java.nio.file.Files
+
+@Requires(TestPrecondition.NOT_WINDOWS)
+class ScalaCompileRelocationIntegrationTest extends AbstractTaskRelocationIntegrationTest {
+
+    private classes = new ScalaCompilationFixture(testDirectory)
+
+    @Override
+    protected String getTaskName() {
+        return ":compileScala"
+    }
+
+    @Override
+    protected void setupProjectInOriginalLocation() {
+        classes.baseline()
+
+        buildFile << classes.buildScript()
+    }
+
+    @Override
+    protected void moveFilesAround() {
+        Files.move(file("src/main/scala").toPath(), file("src/main/new-scala").toPath())
+        classes.sourceDir = 'src/main/new-scala'
+        buildFile.text = classes.buildScript()
+        executer.requireOwnGradleUserHomeDir()
```

Comment:
I want to relocate the `zinc` and `scala` dependencies. The easiest way to do so is to use a different Gradle user home.